### PR TITLE
Allow > inside HTML attributes

### LINKED ISSUE #2913: Media regexes get confused by angle brackets in img tags
https://forums.ankiweb.net/t/images-incorrectly-marked-as-unused-when-checking-media-due-to-special-characters-in-img-tag/38863

Moving from our current regex approach to a HTML parser will likely have a performance hit and increase complexity, so perhaps in the short term we could strip the brackets out of the alt tags when pasting.

### PR DESCRIPTION
**Pull request by vaxr**
Fixes #2913

Extends the `HTML_MEDIA_TAGS` regex to allow for `>` in non-matched attributes. The regex might not deal with things like nested escaped quotes, but neither does the original one, and I didn't want to complicate it even further.

Hope this helps. I'm new here, so sorry if I missed something.

### CODE REVIEW

File: rslib/src/text.rs [hunk 1]
```diff
@@ -104,7 +104,19 @@ lazy_static! {
...
-            <\b(?:img|audio|video|object)\b[^>]+\b(?:src|data)\b=
+            <\b(?:img|audio|video|object)\b
+
+            # any non-`>`, except inside `"` or `'`
+            (?:
+                [^>]
+            |
+                "(?:[^"]+?)"
+            |
+                '(?:[^']+?)'
```
Comment (dae) [inline rslib/src/text.rs:115]: Do we need the inner (?: ... ) in these two clauses?

**Review by dae — COMMENTED** [review]
Thank you for contributing! Just one clarification:

File: rslib/src/text.rs (hunk 1 above)
Comment (vaxr) [reply rslib/src/text.rs]: You're right, we don't need them. Removed.

**Comment by dae** [comment]
Thanks Viktor!

**Comment by vaxr** [comment]
@meliache Thanks for bringing this up! I'll take a look.

**Pull request merged**